enclave-runner/src/loader.rs

```diff
@@ -163,15 +174,34 @@ impl<'a> EnclaveBuilder<'a> {
             signer.miscselect(miscselect, !0);
         }
 
-        let key = PKey::private_key_from_der(include_bytes!("dummy.key")).unwrap();
-        Ok(signer.sign::<_, Hasher>(&*key.rsa().unwrap())?)
+        match self.load_and_sign.take() {
+            Some(f) => f(signer),
+            None => Err(format_err!("either compile with default features or use with_dummy_signature_signer()"))
+        }
     }
 
     pub fn dummy_signature(&mut self) -> &mut Self {
         self.signature = None;
         self
     }
 
+    pub fn with_dummy_signature_signer<H, E, S, T, F>(&mut self, load_key: F)
```

Comment:
Please add a doc comment describing at least the signature of `F`.

I think the appropriate order for the type parameters is H, S, F, E, T. This is based on what I think someone might want to specify vs. what someone might want to have inferred using `_`, with inference-preferred parameters last.